src/js/views/patients/schedule/schedule_views.js

```diff
@@ -230,16 +251,62 @@ const ScheduleListView = CollectionView.extend({
       state: this.state,
     };
   },
-  emptyView: EmptyView,
+  emptyView() {
+    if (this.state.get('searchQuery')) {
+      return EmptyFindInListView;
+    }
+
+    return EmptyView;
+  },
   viewComparator(viewA, viewB) {
     return alphaSort('asc', viewA.model.get('date'), viewB.model.get('date'));
   },
   initialize({ state }) {
     this.state = state;
 
-    this.listenTo(state, {
-      'select:all': this.render,
-      'select:none': this.render,
+    this.listenTo(state, 'change:searchQuery', this.searchList);
+  },
+  searchList(state, searchQuery) {
+    // Must remove the parent view's filter so we have access to previously filtered children
+    this.removeFilter();
+
+    if (!searchQuery) {
+      this.children.each(cv => {
+        cv.removeFilter();
+      });
+      this.setVisibleChildren();
+      return;
+    }
+
+    const matchers = this._buildMatchers(searchQuery);
+
+    this.children.each(cv => {
```

Comment:
Just a quick note about this: Putting this logic on the `DayListView` causes this filter to execute after the `ScheduleListView` filter. We need to ensure this filter happens first so the `ScheduleListView` filter can determine whether or not to hide the day row.